server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/VacationResponseGetMethod.scala

```diff
@@ -0,0 +1,110 @@
+import javax.inject.Inject
+import org.apache.james.jmap.api.vacation.{AccountId, VacationRepository}
+import org.apache.james.jmap.json.Serializer
+import org.apache.james.jmap.mail.VacationResponse.UnparsedVacationResponseId
+import org.apache.james.jmap.mail.{VacationResponse, VacationResponseGetRequest, VacationResponseGetResponse, VacationResponseNotFound}
+import org.apache.james.jmap.model.CapabilityIdentifier.CapabilityIdentifier
+import org.apache.james.jmap.model.Invocation.{Arguments, MethodName}
+import org.apache.james.jmap.model.State.INSTANCE
+import org.apache.james.jmap.model.{CapabilityIdentifier, ErrorCode, Invocation, MissingCapabilityException}
+import org.apache.james.jmap.routes.ProcessingContext
+import org.apache.james.mailbox.MailboxSession
+import org.apache.james.metrics.api.MetricFactory
+import org.reactivestreams.Publisher
+import play.api.libs.json.{JsError, JsObject, JsSuccess}
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+object VacationResponseGetResult {
+  def found(vacationResponse: VacationResponse): VacationResponseGetResult =
+    VacationResponseGetResult(Set(vacationResponse), VacationResponseNotFound(Set.empty))
+  def notFound(vacationResponseId: UnparsedVacationResponseId): VacationResponseGetResult =
+    VacationResponseGetResult(Set.empty, VacationResponseNotFound(Set(vacationResponseId)))
+}
+
+case class VacationResponseGetResult(vacationResponses: Set[VacationResponse], notFound: VacationResponseNotFound) {
+  def merge(other: VacationResponseGetResult): VacationResponseGetResult =
+    VacationResponseGetResult(this.vacationResponses ++ other.vacationResponses, this.notFound.merge(other.notFound))
+}
+
+class VacationResponseGetMethod @Inject() (serializer: Serializer,
+                                           vacationRepository: VacationRepository,
+                                           metricFactory: MetricFactory) extends Method {
+  override val methodName: MethodName = MethodName("VacationResponse/get")
+
+  override def process(capabilities: Set[CapabilityIdentifier],
+                       invocation: Invocation,
+                       mailboxSession: MailboxSession,
+                       processingContext: ProcessingContext): Publisher[Invocation] = {
+    metricFactory.decoratePublisherWithTimerMetricLogP99(JMAP_RFC8621_PREFIX + methodName.value,
+      validate(capabilities)
+        .fold(e => SMono.just(Invocation.error(ErrorCode.UnknownMethod, invocation.methodCallId)),
+          success => asVacationResponseGetRequest(invocation.arguments)
+            .flatMap(vacationResponseGetRequest =>
+              getVacationResponse(vacationResponseGetRequest, processingContext, mailboxSession)
+                .reduce(VacationResponseGetResult(Set.empty, VacationResponseNotFound(Set.empty)), (result1: VacationResponseGetResult, result2: VacationResponseGetResult) => result1.merge(result2))
+                .map(vacationResult => VacationResponseGetResponse(
```

Comment:
This process method is long to me, maybe we can enhance it via method extraction?

Maybe extracting: `vacationResult => VacationResponseGetResponse` (as a method of vaxationResult???) could help